lib/seahorn/BvOpSem2Context.hh

```diff
@@ -17,6 +17,7 @@ namespace details {
 
 // This enumerates the different kinds of metadata memory used in Tracking
 // memory.
+// TODO: move to enum class when int value is not needed.
```

Comment:
where is the implicit cast to underlying type is used?

Reply:
When changing MemValTy to Expression and vice versa